internal/controller/serverset/nic_controller.go

```diff
@@ -49,7 +50,7 @@ func (k *kubeNicController) Create(ctx context.Context, cr *v1alpha1.ServerSet,
 	}
 	lanID := network.Status.AtProvider.LanID
 	// no NIC found, create one
-	createNic := fromServerSetToNic(cr, name, serverID, lanID, replicaIndex, version)
+	createNic := fromServerSetToNic(cr, name, serverID, lanID, serverSetNic, replicaIndex, version)
```

Comment:
You don't need to pass the `serverSetNic` as argument to the function since you already have that information in the `cr` argument. Instead you could extract the `serverSetNic` from the `cr` directly in the `fromServerSetToNic` function.

Reply:
Then I will need to do `cr.Spec.ForProvider.Template.Spec.NICs[nicIndex]` four times, on the lines 41, 154, 159 and 162. Is it more appropriate?